### DISCUSSION TITLE
Using prisma with 'next/router'

### DISCUSSION THREAD

**Discussion by MikeV2000** [prisma/prisma#17192] [Q&A] [answered]
Hi everyone,

I'm trying to use 'useRouter' to get an Id, which I then query for in my database using prisma. The problem I'm having is that I can use 'useRouter' only inside a React Component and Prisma only inside the getServerSideProps function. How can I pass the value from route to the prisma client? 

Thanks in advance.

**Comment by nurul3101** [comment, marked as answer]
Hello @MikeV2000 👋 

Can you pass the value directly to the `getServerSideProps` function as an argument, and then use it with the Prisma client inside the function?

Something like this?
```typescript
export async function getServerSideProps(context) {
  const { id } = context.query;
  const client = new PrismaClient();

  // Use the id value with the client here
}
```
I believe for this you would need to pass the id through routes.

**Reply by MikeV2000** [reply]
Hi, 

thanks for your suggestion. I'm working with Typescript so I had to tweak things a little bit, but you are totally right, that's how to do it. Thank you very much and sorry for asking basically a NextJS question in the Prisma Discurssions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
